text/0000-rust-governance.md

```diff
@@ -0,0 +1,728 @@
+  offender privately. The message should include relevant portions of the CoC
+  together with the offending comments. Again, the goal is to de-escalate, and
+  the email should be written in a dispassionate and empathetic way. However,
+  the message should also make clear that continued violations may result in a
+  ban.
+
+* If problems still persist, the moderators can ban the offender. Banning should
+  occur for progressively longer periods, for example starting at 1 day, then 1
+  week, then permanent. The moderation subteam will determine the precise
+  guidelines here.
+
+In general, moderators can and should unilaterally take the first step, but
+steps beyond that (particularly banning) should be done via consensus with the
+other moderators. Permanent bans require core team approval.
+
+Some situations call for more immediate, drastic measures: deeply inappropriate
+comments, harassment, or comments that make people feel unsafe. (See the
+[code of conduct](http://www.rust-lang.org/conduct.html) for some more details
+about this kind of comment). In these cases, an individual moderator is free to
+take immediate, unilateral steps including redacting or removing comments, or
+instituting a short-term ban until the subteam can convene to deal with the
+situation.
+
+The moderation team is responsible for interpreting the CoC. Drastic measures
+like bans should only be used in cases of clear, repeated violations.
+
+Moderators themselves are held to a very high standard of behavior, and should
+strive for professional and impersonal interactions when dealing with a CoC
+violation. They should always push to *de-escalate*. And they should recuse
+themselves from moderation in threads where they are actively participating in
+the technical debate or otherwise have a conflict of interest. Moderators who
+fail to keep up this standard, or who abuse the moderation process, may be
+removed by the core team.
+
+Subteam, and especially core team members are *also* held to a high standard of
+behavior. Part of the reason to separate the moderation subteam is to ensure
+that CoC violations by Rust's leadership be addressed through the same
+independent body of moderators.
+
+Moderation covers all rust-lang venues, which currently include github
+repos, IRC channels (#rust, #rust-internals, #rustc, #rust-libs), and
+the two discourse forums. (The subreddit already has its own
+moderation structure, and isn't directly associated with the rust-lang
+organization.)
+
+# Drawbacks
+
+One possibility is that decentralized decisions may lead to a lack of coherence
+in the overall design of Rust. However, the existence of the core team -- and
+the fact that subteam leaders will thus remain in close communication on
+cross-cutting concerns in particular -- serves to greatly mitigate that risk.
+
+As with any change to governance, there is risk that this RFC would harm
+processes that are working well. In particular, bringing on a large number of
```

Comment:
"large number" is scary here - I hope the absorption of new people into decision making roles is slow and gradual. It might be worth being explicit in this RFC about the sort of numbers we expect in each subteam initially and (roughly, obvs) how quickly we expect the subteams to grow (for that matter the core team too, there are hints in this RFC that you would like the core team to grow, but aiui there has previously been pressure that it should not grow, or at least not quickly).